test/FIFOMarket.test.ts

```diff
@@ -9,46 +11,99 @@ import {
   createRemovalTokenId,
 } from '@/test/helpers';
 
+interface RemovalDataForListing {
+  amount: number;
+  vintage?: number;
+  supplier?: string;
+}
+
+interface RemovalDataFromListing {
+  listedRemovalIds: BigNumber[];
+  totalAmountOfSupply: number;
+  totalAmountOfSuppliers: number;
+  totalAmountOfRemovals: number;
+}
+
+const parseNumberToBigNumber = (numberToParse: number): string =>
+  hre.ethers.utils.parseUnits(numberToParse.toString()).toString();
+
+const getTotalAmountOfSupply = (removals: RemovalDataForListing[]): number =>
+  removals.reduce((sum, removal) => sum + removal.amount, 0);
+
+const getTotalAmountOfSuppliers = (removals: RemovalDataForListing[]): number =>
+  removals.reduce(
+    (supplierSet, removal) => supplierSet.add(removal.supplier),
+    new Set()
+  ).size ?? 1;
```

Comment:
why is the default 1?

Reply:
Because this would return 0 if removals don't have a supplier defined (which is most of them)

Reply:
why dont they?

Reply:
Because the function that creates the removals on chain is set up to assign supplier to a default address if none is defined, and I decided to rely on that outstanding functionality, rather than explicitly assign a supplier to each and all of these removals (which would have been the default anyway).

I think I might take another pass at some of this testing infrastructure when I start writing more tests, but didn't want to mess around too much in one PR.